pkg/standalone-utils/contracts/LPOracleFactoryBase.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.24;
+
+import { AggregatorV3Interface } from "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import { ILPOracleFactoryBase } from "@balancer-labs/v3-interfaces/contracts/standalone-utils/ILPOracleFactoryBase.sol";
+import { ILPOracleBase } from "@balancer-labs/v3-interfaces/contracts/standalone-utils/ILPOracleBase.sol";
+import { IBasePool } from "@balancer-labs/v3-interfaces/contracts/vault/IBasePool.sol";
+import { IVault } from "@balancer-labs/v3-interfaces/contracts/vault/IVault.sol";
+
+import { InputHelpers } from "@balancer-labs/v3-solidity-utils/contracts/helpers/InputHelpers.sol";
+import { SingletonAuthentication } from "@balancer-labs/v3-vault/contracts/SingletonAuthentication.sol";
+
+abstract contract LPOracleFactoryBase is ILPOracleFactoryBase, SingletonAuthentication {
+    uint256 internal _oracleVersion;
+    mapping(IBasePool => ILPOracleBase) internal _oracles;
+    mapping(ILPOracleBase => bool) internal _isOracleFromFactory;
+
+    constructor(IVault vault, uint256 oracleVersion) SingletonAuthentication(vault) {
+        _oracleVersion = oracleVersion;
+    }
+
+    /// @inheritdoc ILPOracleFactoryBase
+    function create(IBasePool pool, AggregatorV3Interface[] memory feeds) external returns (ILPOracleBase oracle) {
+        if (address(_oracles[pool]) != address(0)) {
```

Comment:
I'm thinking that this could be an issue.
You might create an oracle with the wrong price feeds for a given pool, and this wouldn't allow you to create another one.

We should hash the feeds and the pool to make a unique internal identifier, and maybe disallow creating an oracle for a given pool with the same configuration twice. But more than one oracle for a given pool could be OK if the configuration is not the same.

Reply:
So, after the offline discussion, moving on with the hash of pool and feeds.